defaults.go

```diff
@@ -31,6 +31,14 @@ func dfault(d interface{}, given ...interface{}) interface{} {
 	return given[0]
 }
 
+// Similar to dfault but only considers nil as unset. 0, false, "" are considered set
```

Comment:
noticed a slight type here in dfault

Reply:
the existing method for `default` function is named `dfault`, referring that here

Reply:
Gotcha, I see `default` is a keywork so that makes sense. Carry on!